packages/php-domain/src/PhpDomain/MethodNameService.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3Docs\PhpDomain\PhpDomain;
+
+use Psr\Log\LoggerInterface;
+use T3Docs\PhpDomain\Nodes\MethodNameNode;
+
+class MethodNameService
+{
+    /** https://regex101.com/r/QrrSXk/1 */
+    private const PATTERN = '/^\s*(\w+)\s*\(\s*(.*?)\s*\)\s*(?::\s*(\w+))?\s*$/';
```

Comment:
To be valid with #68 the constant has to be suffed with `_REGEX`. I would also suggest to use another name than `PATTERN`, maybe `METHOD_SIGNATURE_REGEX`?